.github/workflows/e2e.yml

```diff
@@ -0,0 +1,166 @@
+name: SalesforceCommerceCloud/pwa-kit/e2e
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - setup-e2e-testing
+  schedule:
+    # Run every day at 12am (PST) - cron uses UTC times
+    - cron: "0 8 * * *"
+
+jobs:
+  run-generator-demo:
+    strategy:
+      fail-fast: false
+      matrix:
+        # Run all matrix env at once because we will not deploy demo app to MRT.
+        node: [16, 18]
+        npm: [8, 9]
+    runs-on: ubuntu-latest
+    env:
+      # The "default" npm is the one that ships with a given version of node.
+      # For more: https://nodejs.org/en/download/releases/
+      # (We also use this env var for making sure a step runs once for the current node version)
+      IS_DEFAULT_NPM: ${{ (matrix.node == 16 && matrix.npm == 8) || (matrix.node == 18 && matrix.npm == 9) }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          cache: npm
+
```

Comment:
For repetitive steps like this, there's no concept of "partials" where logic can be re-used is there?

Reply:
Not really. Github actions needs the checkout steps to be in the workflow to be able to pull in updated changes from the partials/actions we've defined for the other steps.